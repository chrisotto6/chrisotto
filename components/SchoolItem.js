import kebabCase from '@/lib/utils/kebabCase'

const SchoolItem = ({ schoolName, degree, gpa, awards }) => {
  return (
    <div data-testid={`school-${kebabCase(schoolName)}`}>
      <h3>{schoolName}</h3>
      <h4>{degree}</h4>
      <h5>{`${gpa} - ${awards}`}</h5>
    </div>
  )
}

export default SchoolItem
