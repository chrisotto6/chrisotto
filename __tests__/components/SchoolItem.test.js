import { render } from '@testing-library/react'
import SchoolItem from '@/components/SchoolItem'

it('renders SchoolItem correctly', () => {
  const school = {
    schoolName: 'Lakeland University',
    degree: 'Bachelors of Science - Computer Science',
    gpa: '3.9',
    awards: 'Summa Cum Lade',
  }
  const { container } = render(
    <SchoolItem
      school={school.schoolName}
      degree={school.degree}
      gpa={school.gpa}
      awards={school.awards}
    />
  )
  expect(container).toMatchSnapshot()
})
