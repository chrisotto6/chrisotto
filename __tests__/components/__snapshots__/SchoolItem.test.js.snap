// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SchoolItem correctly 1`] = `
<div>
  <div
    data-testid="school-"
  >
    <h3 />
    <h4>
      Bachelors of Science - Computer Science
    </h4>
    <h5>
      3.9 - Summa Cum Lade
    </h5>
  </div>
</div>
`;
